### ISSUE TITLE
Connection timeouts in case of pod-to-pod communication when both pods are on the same host node

### ISSUE THREAD

**Issue by pbelousov** [Azure/AKS#790]
**What happened**:
Sometimes (one time per several minutes) it's impossible to make HTTP requests between two pods which deployed on the same host node. For about 4-5 seconds services are failing with connection timeout. It occurs only in case of intra-node communication, all requests from other cluster nodes or from outside of the cluster work absolutely well.

**What you expected to happen**:
Obviously, I expect to not have such connection timeout issues.

**How to reproduce it (as minimally and precisely as possible)**:
1. Deploy a service (I tried REST API on .NET Core app with Kestrel) to Node X
2. Deploy a pod with 'curl' tool to exactly the same Node X
3. Run a loop to make requests from the pod with 'curl' to the pod with the REST service:
```while true; do sleep 5; curl -m 1 http://<POD_ID>; echo -e '\n\n'$(date);done ```
4. One time per several minutes this loop returns connection timeout message:
```
Tue Jan 22 09:14:11 UTC 2019
{"status":"OK","version":"18.12.25.2"}

Tue Jan 22 09:14:16 UTC 2019
curl: (28) Connection timed out after 1001 milliseconds


Tue Jan 22 09:14:22 UTC 2019
curl: (28) Connection timed out after 1001 milliseconds


Tue Jan 22 09:14:28 UTC 2019
{"status":"OK","version":"18.12.25.2"}
```

**Anything else we need to know?**:
We run tcpdump for these broken requests and here is the result:

```
15:23:42.323399 IP 100.99.129.63.49436 > 100.99.129.171.http: Flags [S], seq 3203845694, win 29200, options [mss 1460,sackOK,TS val 617065427 ecr 0,nop,wscale 7], length 0
15:23:42.323464 IP 100.99.129.51.49436 > 100.99.129.171.http: Flags [S], seq 3203845694, win 29200, options [mss 1460,sackOK,TS val 617065427 ecr 0,nop,wscale 7], length 0
15:23:42.323893 IP 100.99.129.51.49436 > 100.99.129.171.http: Flags [S], seq 3203845694, win 29200, options [mss 1418,sackOK,TS val 617065427 ecr 0,nop,wscale 7], length 0
15:23:42.323931 IP 100.99.129.51.6506 > 100.99.129.171.http: Flags [S], seq 3203845694, win 29200, options [mss 1418,sackOK,TS val 617065427 ecr 0,nop,wscale 7], length 0

15:23:42.323967 IP 100.99.129.171.http > 100.99.129.51.6506: Flags [S.], seq 1402956205, ack 3203845695, win 28960, options [mss 1460,sackOK,TS val 3992908177 ecr 617065427,nop,wscale 7], length 0
15:23:42.323975 IP 100.99.129.171.http > 100.99.129.51.49436: Flags [S.], seq 1402956205, ack 3203845695, win 28960, options [mss 1460,sackOK,TS val 3992908177 ecr 617065427,nop,wscale 7], length 0

15:23:42.323996 IP 100.99.129.51.6506 > 100.99.129.171.http: Flags [R], seq 3203845695, win 0, length 0
15:23:42.323998 IP 100.99.129.51.6506 > 100.99.129.171.http: Flags [R], seq 3203845695, win 0, length 0
```

In normal cases we don't get this RESET packet, but we don't understand why we have it. It's strange that in case of requests outside the cluster we don't have such issues.

**Environment**:
- Kubernetes version (use `kubectl version`): v1.11.5
- Size of cluster (how many worker nodes are in the cluster?): 2 worker nodes
- General description of workloads in the cluster (e.g. HTTP microservices, Java app, Ruby on Rails, machine learning, etc.): it's mostly .NET Core HTTP microservices with almost no workload as it's DEV cluster

What could we check more in the cluster configuration to identify the root cause of the issue?
Let me know if I need to provide additional info.

Thank you in advance!

**Comment by grampelberg** [comment]
I ran into this on 1.11.5 and fixed it with:

```bash
kubectl -n kube-system delete po -l "component=tunnel"
```

**Comment by pbelousov** [comment]
@grampelberg, thank you for your help! Will try to test this next week.
But as I understand it's some kind of AKS component and if it has some issues I expect that AKS team has to fix it (... or at least respond to this ticket).

**Comment by grampelberg** [comment]
@pbelousov that'll delete the pod which will cause k8s to reschedule it (effectively restarting it).

**Comment by pbelousov** [comment]
The issue was solved after upgrading the cluster to the 1.12.4 version.
